lib/rspec/rails/matchers/active_job.rb

```diff
@@ -230,11 +230,26 @@ def initialize(job)
           def matches?(proc)
             raise ArgumentError, "have_enqueued_job and enqueue_job only support block expectations" unless Proc === proc
 
-            original_enqueued_jobs_count = queue_adapter.enqueued_jobs.count
+            original_enqueued_jobs_hashes = queue_adapter.enqueued_jobs.map(&:hash)
+
             proc.call
-            in_block_jobs = queue_adapter.enqueued_jobs.drop(original_enqueued_jobs_count)
 
-            check(in_block_jobs)
+            in_block_jobs = queue_adapter.enqueued_jobs.each_with_object({}) do |job, jobs|
```

Comment:
It appears to me that storing by the `hash` key is not necessary.
I've added a couple more jobs of the same kind, and they have a different `job_id` assigned, resulting in a different `hash`.

As a result, `count` is always a 1.

Below, a `-= 1` and a further `.positive?` with `Array.new` and a `compact` is basically subtraction of previously enqueued jobs from all enqueued jobs.

I dared pushing a commit that compares with a `Set#-`.

Can you write a failing spec to prove this approach is insufficient?